File: PlayersInitializer.lua
require "HumanPlayer"
require "BotPlayer"

PlayersInitializer = {}

function PlayersInitializer.initialize(game, numHumans)
    local i = 1
    local players = game._players
    while i <= numHumans do
        players[i] = HumanPlayer:new(i)
        i = i + 1
    end

    while i <= players.size do
        players[i] = BotPlayer:new(i, game._cellCoordinateGenerator)
        i = i + 1
    end
end

File: PlayersInitializer.cpp
#include "PlayersInitializer.h"
#include "HumanPlayer.h"
#include "BotPlayer.h"
#include "TicTacToeGame.h"

void PlayersInitializer::initialize(TicTacToeGame* game, int numHumans) {
    int i = 0;
    for (; i < numHumans; ++i) {
        game->m_players[i] = new HumanPlayer(i + 1);
    }
    for (; i < game->m_players.size(); ++i) {
        game->m_players[i] = new BotPlayer(i + 1, game->m_randomCellMaker);
    }
}
